### ISSUE TITLE
Improve ML Model Accuracy to ~70%

### ISSUE THREAD

**Issue by naishasinha** [naishasinha/Fantasy-Basketball#1]
Ideally, industry standards for ML accuracy are usually between 70% and 90%. Currently, this project's base model is at 63% accuracy.

**Comment by naishasinha** [comment]
Next Steps: 
(1) use xgboost and/or random forest classifier
(2) experiment with the different feature numbers
(3) utilize the backwards feature selector rather than forward.

**Comment by naishasinha** [comment]
Course of Action (Dec. 2024): 
***
**Data Cleaning & Feature Engineering**:
* Ensure no key columns (e.g. `fg%` and `won`) have missing values using `.isnull().sum()`, which will return the number of rows excluding the null ones. Impute missing values with median/mean
* Normalize and Standardize Numerical Features: Normalize features like `fga`, `3p%`, `org_max_opp`, and `drag_max_opp` to improve model convergence. To do this, use `StandardScaler` or `MinMaxScaler` from Scikit-learn. We need to do this because the features might have different scales, so normalizing/standardizing them ensures that all features contribute equally to model performance and prevents dominance by larger-scale features.

**Feature Engineering**
* Home vs. Away Advantage: Encode `home_opp` as as binary feature and analyze the impact on `won`. Encoding this will help the model account for a bias that naturally exists in many NBA games.

**Cross-Validation and Test Dataset**
**Metrics and Error Analysis**
